### PR TITLE
:recycle: DOP-3885 consolidate RepoBranchesRepository and RepoBranchesRepository into one class

### PR DESCRIPTION
**Pull request by caesarbell**
This consolidates the two classes into one (`RepoBranchesRepository` and `RepoBranchesRepository`)

We can see the results are successfully [built](https://workerpoolstaging-qgeyp.mongodbstitch.com/pages/job.html?collName=dotcom_queue&jobId=64df86421c12f53a994bb09f) and stored correctly in Atlas. 

image

### CODE REVIEW

File: src/repositories/repoBranchesRepository.ts [hunk 1]
```diff
@@ -8,13 +8,44 @@ export class RepoBranchesRepository extends BaseRepository {
...
   }
 
+  async getRepo(repoName: string): Promise<any> {
+    const query = { repoName: repoName };
+    const repo = await this.findOne(query, `Mongo Timeout Error: Timedout while retrieving branches for ${repoName}`);
+    // if user has specific entitlements
+    return repo;
+  }
+
+  async getRepoBranches(repoName: string): Promise<any> {
```
Comment (rayangler) [inline src/repositories/repoBranchesRepository.ts:42]: With these changes, are we able to safely delete the `branchRepository.ts` file?

Comment (caesarbell) [reply src/repositories/repoBranchesRepository.ts:42]: @rayangler yes, but I just wanted to make sure we were good with the testing, which we are, and I wanted to make sure the feedback was good as well. My plan was to remove it before merging in the code.

Comment (rayangler) [reply src/repositories/repoBranchesRepository.ts:42]: Should be good to remove 👍

**Review by rayangler — APPROVED** [review]
LGTM!

**Review by mmeigs — APPROVED** [review]
LGTM!

**Pull request merged**